subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/LegacyTransformer.java

```diff
@@ -54,7 +65,24 @@ public boolean requiresDependencies() {
         return validateOutputs(primaryInput, outputDir, ImmutableList.copyOf(outputs));
     }
 
+    @Override
+    protected IsolatableParameters<Object[]> doIsolateParameters() {
+        Hasher hasher = Hashing.newHasher();
+        appendActionImplementation(classLoaderHierarchyHasher, hasher, getImplementationClass());
+
+        Isolatable<Object[]> isolatableParameters;
+        try {
+            isolatableParameters = isolatableFactory.isolate(parameters);
```

Comment:
I would probably leave the isolation where it is for legacy transforms, as it is is a breaking change to do this at execution time instead of at registration.

Reply:
Done.